wordpress-mobile/WordPress-iOS
# [Site Name] Add Site Name screen to the Site Creation flow

Fixes #18299

This PR adds the Site Name screen to the Site Creation Flow.

#### Note ####
@twstokes , I know that you opened a [PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/18318) for setting the vertical name in the Site Name screen, but since with this approach that basically came for free, I thought I'd add it in here, please let me know if this is ok, otherwise we can certainly handle it in a separate PR.

### To test: ###

#### Pre-requisite ####
in order to properly test this PR, you might want to bypass the checks of the `siteName` feature flag, as well as the A/B test. The easiest way to do so is to change the following code, around line 12 in `SiteCreationWizardLauncher`:

```
   private var shouldShowSiteIntent: Bool {
        return intentVariant == .treatment && FeatureFlag.siteIntentQuestion.enabled
    }

    private var shouldShowSiteName: Bool {
        return nameVariant == .treatment(nil) && FeatureFlag.siteName.enabled
    }
```

by temporarily setting both `shouldShowSiteIntent` and `shouldShowSiteName` to true.

#### Testing steps ####

- checkout, apply the pre-requisite, build and run this branch
- Go to My Site, then start the site creation flow
- Tap on any available vertical, or enter a custom one and tap on it
- Verify that the Site Name screen is shown, and that the title vertical name is displayed in blue in the header (see animation below)
- Go back and tap on "Skip", and verify that the title is "Give your website a name" without extra spaces in it
- Enter a site name, then tap continue
- Verify that the design selection screen is shown
- Go back and verify that you can see the Site Name screen
- Tap on "Skip" and make sure the Design Selection screen is shown

<p align=center>
<img height=500 src=https://user-images.githubusercontent.com/34376330/162647144-b80e1ff8-18c9-4ebe-a0a6-7ee05b8a179c.gif>
</p>


## Regression Notes
1. Potential unintended areas of impact
None, this change is self-contained inside the Site Name classes

2. What I did to test those areas of impact (or what existing automated tests I relied on)
NA

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:
